### DISCUSSION TITLE
In a DOCX file, lists are converted to Compact-type paragraphs

### DISCUSSION THREAD

**Discussion by ptram** [quarto-dev/quarto-cli#9246] [Q&A]
### Description

Hi,

When converting a Markdown document to DOCX, using the standard DOCX template generated by Quarto, lists are converted to Compact-type paragraphs. This is both with bulleted and numbered lists.

For example, this code:

`* This is just an example of a bullet point.
`

or this:

`1. And this is an example of a numbered point.`

are converted to paragraphs with the Compact paragraph style applied. A bullet is prefixed to the first example, and the shown number to the second one.

<img width="186" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/19186065/a76a2e7f-c113-44b1-908a-9c2c5e903ca1">

and

<img width="200" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/19186065/cb66a9cc-9ef4-4013-b039-052723ee6beb">

I know that I can create the relevant fenced styles for Bulleted and Numbered paragraphs, but I wonder if – being these very basic styling – something is already available, and I'm just doing something wrong with my syntax.

**Comment by mcanouil** [comment]
@ptram it would be appreciated for the future if you could make your example more complete and using code blocks to make it easy for everyone to use/help.

This being said that's Pandoc:

```bash
echo "* This is just an example of a bullet point. " > index.qmd
quarto pandoc index.qmd --from markdown --to docx -o index.docx
```

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by ptram** [comment]
@mcanouil very sorry, but I'm very primitive in using GitHub.

This is an example of the code:

````
---
title: A test about lists
author: Me Myself
format:
  docx:
    reference-doc: custom-reference-doc.docx
    lang: it
    dir: ltr
---
  
This is a bulletted list:

* A first item.
* A second item.

And this is a numbered list:

1. This is the first one.
2. This is the second one.
3. And this is the third.

All the above have the Compact paragraph style assigned.
````

You can also find the DOCX template in the attached zip file, together with the Markdown code and the resulting DOCX output.

If I understand correctly from your answer, I'm doing the right thing with lists. Only, I wonder why the Compact paragraph style is assigned to the lists, instead of a dedicated Numbered and Bulleted style.

[lists-test.zip](https://github.com/quarto-dev/quarto-cli/files/14828366/lists-test.zip)

This is my quarto check output:

----------------

Quarto 1.4.551
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.551
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /Library/TeX/texbin
      Version: 2022

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.4
      Path: /Library/Frameworks/Python.framework/Versions/3.11/bin/python3
      Jupyter: 5.7.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/

--------------

**Reply by mcanouil** [reply]
No need to apologise.
Note that there is nothing specific to GiHub about reproducible example. You'll see the exact same on StackExchange/Overflow, Reddit, etc.

Note also, that writing on GitHub is no different from writing a Quarto document in Markdown. Here you did not properly open or close your code block.

Regarding your issue, my reply did contain the answer in addition to the request to provide complete code *in the future*.
It's Pandoc that does the formatting that way.

**Reply by ptram** [reply]
By examining the resulting DOCX file, I see that paragraph style and numbering style are kept separate. Therefore, there are two styles acting on the lists.

I've found elsewhere that the Compact paragraph style corresponds to a compact type of list, where there is little space between the items.

Paolo

**Reply by mcanouil** [reply]
FYI, https://pandoc.org/chunkedhtml-demo/8.7-lists.html

**Comment by cderv** [comment]
If you use the loose format to create the list, then you'll get Normal as paragraph style in default template
````markdown
---
title: A test about lists
author: Me Myself
format: docx
---
  
This is a bulletted list:

* A first item.
* A second item.

And this is a numbered list:

1. This is the first one.
2. This is the second one.
3. And this is the third.

All the above have the Compact paragraph style assigned.

This is a bulletted list in loose form:

* A first item.

* A second item.

And this is a numbered list in loose form:

1. This is the first one.

2. This is the second one.

3. And this is the third.

All the above have the Normal paragraph style assigned.

````

Note the two ways to create lists (as shown in the doc share in linked in previous thread)

**Reply by ptram** [reply]
Thank you. The best solution, for what I want to do (customizing the format of lists and the correlated paragraphs) is to create custom styles. I would therefore have something like this for bulleted lists:

````
::: {custom-style="Bulleted"}
* A first item.
* A second item.
:::

````

And this for numbered lists:


````
::: {custom-style="Numbered1"}
1. A first item.
:::
::: {custom-style="Numbered"}
2. A second item.
3. A third item.
:::

````

**Reply by joelnitta** [reply]
@ptram

Were you successful in doing this with a custom style (Word doc)?

I have been trying to create one but am not able to get it to work.
